pkg/controller/job/job_controller_test.go

```diff
@@ -1993,6 +2002,522 @@ func TestSyncJobDeleted(t *testing.T) {
 	}
 }
 
+func TestSyncJobWithJobPodFailurePolicy(t *testing.T) {
```

Comment:
Do you get 100% coverage of `getStatus` with these testcases?

Reply:
Yes, but running the coverage made me realize the unit tests do cover the new code in `trackJobStatusAndRemoveFinalizers` due to the broken test setup. Fixed, now both places are covered.